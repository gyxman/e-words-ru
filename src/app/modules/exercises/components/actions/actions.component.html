<button
    automation-id="submit-button"
    mat-raised-button
    color="primary"
    class="button"
    (click)="onSubmit()"
    [disabled]="disabledSubmit"
>
    Проверить
</button>
