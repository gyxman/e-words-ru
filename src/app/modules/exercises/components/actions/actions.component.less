:host {
    display: block;
    max-width: 200px;
}

.button {
    width: 100%;
}
