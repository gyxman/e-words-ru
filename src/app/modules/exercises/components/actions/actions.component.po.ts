import {PageObject} from '../../../../testing/page-object';
import {DebugElement} from '@angular/core';

export class ActionsComponentPo<T> extends PageObject<T> {
    get submitButton(): DebugElement {
        return this.getByAutomationId('submit-button');
    }
}
