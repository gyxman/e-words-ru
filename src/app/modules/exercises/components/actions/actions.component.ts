import {
    ChangeDetectionStrategy,
    Component,
    EventEmitter,
    Input,
    Output,
} from '@angular/core';

@Component({
    selector: 'app-actions',
    templateUrl: './actions.component.html',
    styleUrls: ['./actions.component.less'],
    changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ActionsComponent {
    @Input() disabledSubmit = false;

    @Output() submit = new EventEmitter();

    onSubmit() {
        this.submit.emit();
    }
}
