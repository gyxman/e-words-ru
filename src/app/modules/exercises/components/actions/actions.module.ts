import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {ActionsComponent} from './actions.component';
import {MatButtonModule} from '@angular/material/button';

@NgModule({
    imports: [CommonModule, MatButtonModule],
    declarations: [ActionsComponent],
    exports: [ActionsComponent],
})
export class ActionsModule {}
