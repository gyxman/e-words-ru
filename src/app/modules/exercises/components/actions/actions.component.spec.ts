import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {ActionsComponent} from './actions.component';
import {MatButtonModule} from '@angular/material/button';
import {Component} from '@angular/core';
import {ActionsComponentPo} from './actions.component.po';

@Component({
    template:
        '<app-actions [disabledSubmit]="disabled" (submit)="submit()"></app-actions>',
})
class TestComponent {
    disabled = false;

    submit() {}
}

describe('ActionsComponent - компонент с кнопками для упражнений', () => {
    let testComponent: TestComponent;
    let fixture: ComponentFixture<TestComponent>;
    let pageObject: ActionsComponentPo<TestComponent>;

    beforeEach(async(() => {
        TestBed.configureTestingModule({
            declarations: [TestComponent, ActionsComponent],
            imports: [MatButtonModule],
        }).compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(TestComponent);
        testComponent = fixture.componentInstance;
        pageObject = new ActionsComponentPo(fixture);
    });

    it('Если компонент загрузился, то все элементы отрисовались', () => {
        // act
        fixture.detectChanges();

        // assert
        expect(fixture.nativeElement).toMatchSnapshot();
    });

    it('Если пришла информация о блокировке кнопки, то блокируем кнопку отправить', () => {
        // arrange
        testComponent.disabled = true;

        // act
        fixture.detectChanges();

        // assert
        expect(pageObject.submitButton.attributes.disabled).toBe('true');
    });

    it('Если информация о блокировке кнопки не пришла, то не блокируем кнопку отправить', () => {
        // arrange
        testComponent.disabled = false;

        // act
        fixture.detectChanges();

        // assert
        expect(pageObject.submitButton.attributes.disabled).toBeNull();
    });

    it('Если пользователь нажимает на кнопку, то эммитим экшен', () => {
        // arrange
        jest.spyOn(testComponent, 'submit');

        fixture.detectChanges();

        // act
        pageObject.click(pageObject.submitButton);

        // assert
        expect(testComponent.submit).toHaveBeenCalledTimes(1);
    });
});
