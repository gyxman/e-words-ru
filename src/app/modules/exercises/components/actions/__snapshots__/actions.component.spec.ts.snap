// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsComponent - компонент с кнопками для упражнений Если компонент загрузился, то все элементы отрисовались 1`] = `
<div
  id="root0"
>
  <app-actions>
    <button
      automation-id="submit-button"
      class="button mat-focus-indicator mat-raised-button mat-button-base mat-primary"
      color="primary"
      mat-raised-button=""
    >
      <span
        class="mat-button-wrapper"
      >
         Проверить

      </span>
      <div
        class="mat-button-ripple mat-ripple"
        matripple=""
      />
      <div
        class="mat-button-focus-overlay"
      />
    </button>
  </app-actions>
</div>
`;
